### ISSUE TITLE
Ordering by percentile not working anymore

### ISSUE THREAD

**Issue by everatala** [elastic/elasticsearch#89079]
### Elasticsearch Version

8.3.2

### Java Version

_bundled_

### OS Version

-

### Problem Description

The percentile ordered graphs show up empty.

### Steps to Reproduce

- Have an index pattern that contains different datapoints: some with 'field01' (number field, in this case, loading times) and some without. 
- Create a Lens visualization of the following: have the 95th percentile of loading times per host, in descending order.

**Observed:**

- The graph is empty, if Other field is added, that contains data points.
- If the number of shown points is big enough, there is an empty part in the beginning, and the visualisation shows up after.

**Expected:**

- 95th percentile of loading times per host shows up

**Note**: Seems like the algorithm puts the empty data in the beginning of the ordered list, instead of the end or skipping it.
See more: [https://discuss.elastic.co/t/ordering-by-percentile-not-working-anymore/308995](https://discuss.elastic.co/t/ordering-by-percentile-not-working-anymore/308995)

**Comment by everatala** [comment]
Our elastic instance is run on AWS and the Frankfurt (es-central-1) node.

**Comment by mayya-sharipova** [comment]
@everatala Thanks for submitting this issue, but this repository is reserved for `elasticsearch` issues, but your issue seems to be in Kibana. Please create an issue in https://github.com/elastic/kibana.

**Comment by everatala** [comment]
Hi @mayya-sharipova ! According to the source I linked, one of the people on the elastic stack forum routed me here, because this issue occurs because the data is coming incorrectly from elasticsearch. Please see [here](https://discuss.elastic.co/t/ordering-by-percentile-not-working-anymore/308995/6):
Quote from Joe Reuter: "Hi VeraE, sorry for the late reply. It seems like the data is shipped from Elasticsearch like this - could you open a bug report here? [GitHub - elastic/elasticsearch: Free and Open, Distributed, RESTful Search Engine](https://github.com/elastic/elasticsearch)"

**Comment by iverase** [comment]
Yes, @everatala I think it is a valid issue. I will reopen as queries are in the discuss issue.
